gcore/src/exec.rs

```diff
@@ -26,9 +26,35 @@ use crate::{
     ActorId, MessageId, ReservationId,
 };
 use gsys::{
-    BlockNumberWithHash, ErrorWithBlockNumberAndValue, ErrorWithGas, ErrorWithHash, HashWithValue,
+    BlockNumberWithHash, ErrorWithBlockNumberAndValue, ErrorWithGas, ErrorWithHash, Gas,
+    HashWithValue, Value,
 };
 
+/// Current version of execution settings.
+///
+/// Backend maintains backward compatibility with previous versions of execution
+/// settings. This structure matches to the most recent version of execution
+/// settings supported by backend.
+#[repr(C, packed)]
+#[derive(Debug, Default, Clone, Copy)]
+pub struct Settings {
```

Comment:
I'd move structure into `gsys` and use it in `gear-core` and `gcore`

Reply:
I haven't done it on purpose as we are going to have multiple versions of this struct on the backend side whereas there should be a single one on the userspace side. I don't think we need any confusing on the latter.

Reply:
That's been done